### ISSUE TITLE
Constant noise assumption in sampling script

### ISSUE THREAD

**Issue by EiffL** [CosmoStat/jax-lensing#43]
If I read this correctly, it looks like we are assuming constant noise on gamma throughout the field in the likelihood term:
https://github.com/CosmoStat/jax-lensing/blob/13a4b8d13deb6d4be1ce62cf7b35974fb3e733be/scripts/sample_maps.py#L213

But we know we have some variations, and that not accounting for it can throw off a model like DeepMass. If I read this part of the code correctly: 
https://github.com/CosmoStat/jax-lensing/blob/13a4b8d13deb6d4be1ce62cf7b35974fb3e733be/scripts/train_deepmass.py#L101
DeepMass uses the non-constant noise variance, but not our DSM sampling? That would be inconsistent and we should fix that.

@b-remy can you let me know if I missed something?

**Comment by b-remy** [comment]
Right, with the last PR, the likelihood will be always using the diagonal covariance matrix. Thanks for spotting this @EiffL
